includes/WpMinions/RabbitMQ/Client.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace WpMinions\RabbitMQ;
+
+use WpMinions\Client as BaseClient;
+
+
+/**
+ * The Gearman Client uses the libGearman API to add jobs to the Gearman
+ * Queue. The servers that the client should connect to are setup as
+ * part of the initialization.
+ */
+class Client extends BaseClient {
+
+	/**
+	 * Instance of Connection class
+	 *
+	 * @var Connection
+	 */
+	public $connection = null;
+
+	/**
+	 * Setup backend
+	 */
+	public function register() {
+		// Do nothing
+	}
+
+	/**
+	 * Connect to host and channel
+	 */
+	private function connect() {
+		if ( null !== $this->connection ) {
+			return $this->connection;
+		}
+
+		try {
+			$this->connection = new Connection();
+		} catch ( \Exception $e ) {
+			return false;
+		}
+
+		return $this->connection;
+	}
+
+	/**
+	 * Adds a Job to the libGearman Client's Queue.
+	 *
+	 * @param string $hook The action hook name for the job
+	 * @param array $args Optional arguments for the job
+	 * @param string $priority Optional priority of the job
+	 * @return bool true or false depending on the Client
+	 */
+	public function add( $hook, $args = array(), $priority = 'normal' ) {
+		if ( ! $this->connect() ) {
+			return false;
+		}
+
+		$job_data = array(
+			'hook'    => $hook,
+			'args'    => $args,
+			'blog_id' => get_current_blog_id(),
+		);
+
+		$message = new \PhpAmqpLib\Message\AMQPMessage(
+			json_encode( $job_data ) );
+
+		$this->connection->get_channel()->basic_publish( $message, '', 'wordpress' );
```

Comment:
@tlovett1 is the `wordpress` value here what would be used to identify jobs in a particular "queue"? If so, we should make it something more unique (we already do this for the gearman connector).

For the gearman connector, we use `WP_Async_Task` concatenated with an optional `WP_ASYNC_TASK_SALT` to generate this unique queue name - this way, if multiple WordPress sites are using the same queue server, we can distinguish between jobs for the different WordPress installations.

Here's how we do it on the gearman implementation: https://github.com/10up/WP-Minions/blob/master/includes/WpMinions/Gearman/Client.php#L120

There isn't really anything specific to Gearman in that `get_async_group` function, so we could probably just move that method wholesale to the parent class.